Update to .net 6.0

PhantomGamers opened the pull request:
- Removes Ionic.Zip.dll in favor of using the DotNetZip nuget package
- Uses System.Text.Json instead of System.Web.Script.Serialization
- Added .editorconfig from https://github.com/dotnet/roslyn/blob/7d69096bedcf8b0b6bad5f5a493659bd46a62895/.editorconfig
- Ran Code Cleanup from VS
- Ran `dotnet format style --severity info`

Can be compiled into a single exe using:
`dotnet publish -p:PublishSingleFile=true -r win-x86 -c Release --self-contained false`

PhantomGamers committed:
Upgrade to .net 6.0

- Removes Ionic.Zip.dll in favor of using the DotNetZip.Reduced nuget package
- Uses System.Text.Json instead of System.Web.Script.Serialization

PhantomGamers committed:
Switch to DotNetZip from DotNetZip.Reduced

DotNetZip.Reduced is outdated and still relies on .net framework

PhantomGamers committed:
Cleanup

- Added .editorconfig from https://github.com/dotnet/roslyn/blob/7d69096bedcf8b0b6bad5f5a493659bd46a62895/.editorconfig
- Ran Code Cleanup from VS
- Ran `dotnet format style --severity info`

PhantomGamers committed:
Update README.md

PhantomGamers committed:
Switch to GitHub's .gitignore for VS

+ include rider's required files for rider support

PhantomGamers committed:
Fix update checker

Dotnet core defaults to UseShellExecute being false which breaks Process.Start's ability to open links without explicitly enabling it.

PhantomGamers committed:
wrap long line

PhantomGamers committed:
Use SDK properties instead of AssemblyInfo

henrikx commented:
Hi, this looks great, thanks for doing this!

Do you think we could remove all external dependencies by using the System.IO.Compression methods for ZipFile extraction? https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.zipfile.extracttodirectory?view=net-6.0

PhantomGamers commented:
It would be possible but it is a similar issue to switching to HttpClient in that .net's ZipFile does not provide an easy way to monitor progress like ionic's does.

PhantomGamers committed:
Remove CodePages dependency

apparently this wasn't required after all?

I was getting an encoding error earlier when I added this to solve it, but I just tested without this dependency and it continued to work so something else I changed must have solved it.

PhantomGamers commented:
I just removed the dependency on System.Text.Encoding.CodePages as it seems it is not needed.

Earlier I was getting a `''IBM437" is not a supported encoding name` error, and I thought adding that solved it but I guess it was something else I changed that wound up solving it.

Now the single exe is smaller than 1.4.16 (424KB vs 1.4.16's 530KB).

(Though you should check to make sure it works on your end as well)
